src/EventStore.Projections.Core/Services/Processing/EmittedStream.cs

```diff
@@ -695,12 +698,30 @@ yield return
 				return null;
 			var failed = topAlreadyCommitted.Item1 != eventsToWrite.CausedByTag ||
 			             topAlreadyCommitted.Item2 != eventsToWrite.EventType;
+			if (failed && eventsToWrite.EventType.Equals(LinkEventType)) {
+				// We check if the linked event still exists. If not, we skip that emitted event.
+				var parts = eventsToWrite.Data.Split(LinkToSeparator, 2);
+				var streamId = parts[1];
+				if (!long.TryParse(parts[0], out long eventNumber))
+					throw new Exception($"Unexpected exception: Emitted event is an invalid link event: Body ({eventsToWrite.Data}) CausedByTag ({eventsToWrite.CausedByTag}) StreamId ({eventsToWrite.StreamId})");
+
+				var source = new TaskCompletionSource<ClientMessage.ReadEventCompleted>();
+				var task = source.Task;
+				_ioDispatcher.ReadEvent(streamId, eventNumber, _writeAs, res => {
+					source.TrySetResult(res);
+				});
+
+				task.Wait(TimeSpan.FromSeconds(5)); // FIXME - Finds better timeout number.
```

Comment:
What should be a proper timeout value in this case? 5 seconds feels way too high.